crates/bevy_ui/src/render/mod.rs

```diff
@@ -276,6 +279,17 @@ impl RenderGraphNode for RunUiSubgraphOnUiViewNode {
     }
 }
 
+pub fn clear_extracted_ui_nodes(
+    mut commands: Commands,
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+) {
+    extracted_uinodes
+        .uinodes
+        .drain()
+        .for_each(|(entity, _)| commands.entity(entity).despawn());
```

Comment:
Isn't it enough to just clear the buffer here? They are all `TemporaryRenderEntity` entites spawned during extraction so there are shouldn't be any collisions and they'll be despawned automatically?

Reply:
I did this because I was worried that the entities spawned might still be mistakenly used in the next stages.
Each of those entity has some component attached to it.
In the `Render` schedule a user can query for the `Component` instead of accessing the entities using the list in the resource. In that case they will get both the current as well as the outdated entities.

Reply:
IIRC I think they just get a `UiBatch` added to them and we only ever look them up individually by their render entity id,  not by querying for the component.